src/adagrams.js

```diff
@@ -1,15 +1,112 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterpool = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  let letters = [];
+  for (let letter of Object.keys(letterpool)) {
+    for (let i = 0; i < letterpool[letter]; i++) {
+      letters.push(letter);
+    }
+  }
+  let hand = [];
+  while (hand.length < 10) {
+    let randomIndex = Math.floor(Math.random() * letters.length);
+    hand.push(letters[randomIndex]);
+    letters.splice(randomIndex, 1);
+  }
+  return hand;
 };
-
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (let char of input) {
+    const index = lettersInHand.indexOf(char);
+    if (index < 0) {
+      return false;
+    } else {
+      lettersInHand.splice(index, 1);
```

Comment:
We do not want to remove letters from `lettersInHand`. This function should only check to see if all of the letters of `input` are in `lettersInHand`. If we are removing letters every time we do a simple check, the user will run out of letters very quickly.